### PR TITLE
NR-38533: Workloads statusConfig field 

### PR DESCRIPTION
**Pull request by NSSPKrishna**
# Description

Workloads update.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

Please delete options that are not relevant.

- [x] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic#testing) for instructions on running tests locally.

## How to test this change?

Please describe how to test your changes. Include any relevant steps in the UI, HCL file(s), commands, etc

- Step 1
- Step 2
- etc

### CODE REVIEW

File: newrelic/resource_newrelic_workload.go [hunk 1]
```diff
@@ -62,6 +69,106 @@ func resourceNewRelicWorkload() *schema.Resource {
...
+										Type:        schema.TypeSet,
+										Required:    true,
+										Description: "The input object used to represent a rollup strategy.",
+										Elem:        WorkloadremainingEntitiesRuleSchemaElem(),
+									},
+								},
+							},
+						},
+
+						"rules": {
```
Comment (kidk) [inline newrelic/resource_newrelic_workload.go:100]: Change to `rule`

File: newrelic/resource_newrelic_workload.go [hunk 2]
```diff
@@ -86,14 +193,58 @@ func resourceNewRelicWorkload() *schema.Resource {
...
+			Optional:    true,
+			Description: "Threshold value defined for the rule. This optional field is used in combination with thresholdType. If the threshold type is null, the threshold value will be ignored.",
+		},
+	}
+}
+
 func resourceNewRelicWorkloadCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*ProviderConfig).NewClient
 	createInput := expandWorkloadCreateInput(d)
 	accountID := d.Get("account_id").(int)
```
Comment (mbazhlekova) [inline newrelic/resource_newrelic_workload.go:243]: We should use `selectAccountID` function here so user doesn't have to give account ID on the resource. See example - https://github.com/newrelic/terraform-provider-newrelic/blob/main/newrelic/resource_newrelic_alert_channel.go#L281

File: newrelic/resource_newrelic_workload_test.go [hunk 3]
```diff
@@ -122,6 +122,159 @@ func TestAccNewRelicWorkload_EntityScopeAccountsOnly(t *testing.T) {
...
+	resourceName := "newrelic_workload.foo"
+	rName := acctest.RandString(5)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckNewRelicWorkloadDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNewRelicWorkloadBasicConfigOnly(rName),
```
Comment (mbazhlekova) [inline newrelic/resource_newrelic_workload_test.go:136]: We should add an update and import step for at least one of these tests. Here's an example - https://github.com/newrelic/terraform-provider-newrelic/blob/main/newrelic/resource_newrelic_alert_policy_channel_test.go#L16

File: testing/newrelic.tf [hunk 4]
```diff
@@ -1,7 +1,7 @@
 terraform {
   required_providers {
     newrelic = {
-      source = "newrelic/newrelic"
+      source  = "newrelic/newrelic"
```
Comment (mbazhlekova) [inline testing/newrelic.tf:4]: Can remove this change

**Review by mbazhlekova — CHANGES_REQUESTED** [review]

File: newrelic/resource_newrelic_workload.go [hunk 5]
```diff
@@ -103,30 +255,31 @@ func resourceNewRelicWorkloadCreate(ctx context.Context, d *schema.ResourceData,
...
 	client := meta.(*ProviderConfig).NewClient
-
 	ids, err := parseWorkloadIDs(d.Id())
 	if err != nil {
 		return diag.FromErr(err)
 	}
-
-	workload, err := client.Workloads.GetWorkloadWithContext(ctx, ids.AccountID, ids.GUID)
+	workload, err := client.Workloads.GetWorkloadWithContext(ctx, ids.AccountID, string(ids.GUID))
 	if err != nil {
```
Comment (kidk) [inline newrelic/resource_newrelic_workload.go:270]: Double err != nil check

File: newrelic/resource_newrelic_workload.go [hunk 6]
```diff
@@ -103,30 +255,31 @@ func resourceNewRelicWorkloadCreate(ctx context.Context, d *schema.ResourceData,
...
-		if _, ok := err.(*errors.NotFound); ok {
-			d.SetId("")
-			return nil
-		}
+		if err != nil {
+			if _, ok := err.(*errors.NotFound); ok {
```
Comment (kidk) [inline newrelic/resource_newrelic_workload.go:272]: Let's validate if this works. Delete the workload from the UI should trigger this.

**Pull request closed without merging**